tests/test_neighborhood.py

```diff
@@ -111,121 +173,149 @@ def test_weighted_center(self):
+        n.coordinate_data.increment_visit_count(Coordinate([1, 1, 1]))
 
-        weighted_center = n._determine_weighted_coordinate_center(
-            coordinates, weights)
+        n.coordinate_data.set_measurement(Coordinate([2, 1, 1]), rcm1)
+        n.coordinate_data.increment_visit_count(Coordinate([2, 1, 1]))
 
-        # Weighted center
-        #   First multiply each coordinate by its weights:
-        #      [4,0,0], [0,1,0], [0,0,4]
-        #   Then sum up the coordinates:
-        #      [4,1,4]
-        #   Then divide by sum of weights (2+1+4=7)
-        #      [4/7, 1/7, 4/7]
-        expected_weighted_center = Coordinate([4 / 7, 1 / 7, 4 / 7])
+        n.coordinate_data.set_measurement(Coordinate([1, 2, 1]), rcm2)
+        n.coordinate_data.increment_visit_count(Coordinate([1, 2, 1]))
 
-        self.assertEqual(weighted_center, expected_weighted_center)
+        step_vector = n._get_step_vector()
+        expected_step_vector = Coordinate([3/4, 1/2, 0])
+        self.assertEqual(step_vector, expected_step_vector)
 
     def test_calculate_new_coordinate_one_dimension(self):
-        """ 
+        """
         Test calculate_new_coordinate for a case where only
-        one of the dimensions increases the throughput
+        one of the dimensions increases the measurement.
         """
-        cd = CoordinateData()
-        cd.set_throughput(Coordinate([0, 0]), 2)
-        cd.set_throughput(Coordinate([1, 0]), 4)
-        cd.set_throughput(Coordinate([0, 1]), 2)
-
         dims = SearchDimensions()
         dims.add_dimensions(0, [
             SearchDimension("foo", SearchDimension.DIMENSION_TYPE_LINEAR),
             SearchDimension("bar", SearchDimension.DIMENSION_TYPE_EXPONENTIAL)
         ])
 
         nc = NeighborhoodConfig(dims, radius=2, min_initialized=3)
+        n = Neighborhood(nc, home_coordinate=Coordinate([0, 0]))
 
-        n = Neighborhood(nc, cd, Coordinate([0, 0]))
+        rcm0 = self._construct_rcm(throughput=1, latency=5)
+        rcm1 = self._construct_rcm(throughput=3, latency=5)
+        rcm2 = self._construct_rcm(throughput=1, latency=5)
 
-        magnitude = 20
-        new_coord = n.calculate_new_coordinate(magnitude)
+        n.coordinate_data.set_measurement(Coordinate([0, 0]), rcm0)  # home coordinate
+        n.coordinate_data.increment_visit_count(Coordinate([0, 0]))
 
-        self.assertEqual(new_coord, Coordinate([3, 0]))
+        n.coordinate_data.set_measurement(Coordinate([1, 0]), rcm1)
+        n.coordinate_data.increment_visit_count(Coordinate([1, 0]))
+
+        n.coordinate_data.set_measurement(Coordinate([0, 1]), rcm2)
+        n.coordinate_data.increment_visit_count(Coordinate([0, 1]))
+
+        magnitude = 20
+        new_coord = n.calculate_new_coordinate(magnitude,
+                                               disable_clipping=True)
+        self.assertEqual(new_coord, Coordinate([10, 0]))
```

Comment:
immediately after try with clipping enabled

Reply:
Added a separate testing for clipping.